mdsuite/graph_modules/molecular_graph.py

```diff
@@ -60,143 +70,141 @@ def __init__(
+                "this information."
+            )
+            raise ValueError(error_msg)
+
+        self._get_molecular_mass()
+        self._build_molecule_groups()  # populate the class group attribute
+        self._perform_isomorphism_tests()  # run the graph tests.
+
+    def _get_molecular_mass(self):
         """
-        Enforce a cutoff on a tensor
+        Get the mass of a SMILES molecule based on experiment data.
 
-        Parameters
-        ----------
-        tensor : tf.Tensor
-        cutoff : float
-        """
+        Returns
+        -------
+        Updates the following class attributes:
 
-        cutoff_mask = tf.cast(
-            tf.less(tensor, cutoff), dtype=tf.int16
-        )  # Construct the mask
+        mass : float
+                mass of the molecule
+        """
+        mass = 0.0
+        for item in self.species:
+            mass += self.experiment.species[item]["mass"][0] * self.species[item]
 
-        return tf.linalg.set_diag(cutoff_mask, np.zeros(len(tensor)))
+        self.molecular_mass = mass
 
-    def build_configuration_graph(self, cutoff: float, adjacency: bool = True):
+    def build_configuration_graph(self) -> tf.Tensor:
         """
         Build a graph for the configuration.
 
-        Parameters
-        ----------
-        cutoff : float
-        adjacency : bool
-                If true, the adjacent matrix is returned.
         Returns
         -------
-
+        adjacency_matrix : tf.Tensor
+                An adjacency matrix for the configuration describing which atoms are
+                bonded to which others.
         """
-        path_list = [join_path(species, "Positions") for species in self.species]
-        data_dict = self.database.load_data(path_list=path_list, select_slice=np.s_[:, 0])
+        path_list = [
+            join_path(species, self.reference_property) for species in self.species
+        ]
+        data_dict = self.database.load_data(
+            path_list=path_list, select_slice=np.s_[:, self.reference_configuration]
+        )
         data = []
         for item in path_list:
             data.append(data_dict[item])
         configuration_tensor = tf.concat(data, axis=0)
-        distance_matrix = self.get_neighbour_list(
+        distance_matrix = get_neighbour_list(
             configuration_tensor, cell=self.experiment.box_array
         )
 
-        return self._apply_system_cutoff(distance_matrix, cutoff)
+        return _apply_system_cutoff(distance_matrix, self.cutoff)
 
-    @staticmethod
-    def reduce_graphs(adjacency_matrix: tf.Tensor, n_molecules: int = None):
+    def _build_molecule_groups(self):
+        """
+        Build molecule groups from decomposed graph.
+
+        Returns
+        -------
+
+        """
+        adjacency_graph = self.build_configuration_graph()
+        decomposed_graphs = self._perform_graph_decomposition(adjacency_graph)
+        self.molecular_groups = self._split_decomposed_graphs(decomposed_graphs)
```

Comment:
This method only calls three other methods an has no arguments and is only called once so I would vote to remove it.

Reply:
I vote to keep it as it gives the 3 operations a common name

Reply:
agreed